### PR TITLE
Refurbishing form elements

### PR DESCRIPTION
**Pull request by salman0719**
- `useDefaultValue` hook: to apply loaded value into form inputs without reloading the entire form
- Abstracting `TextBox` & `TextArea` rendering mechanism into a new component
- `TextBox`, `TextArea`, `Dropdown`, `CheckBox`, `Radio`, `Pill`, `CloseButton` - `disabled` prop
- Synchronizing Profile forms/inputs with new formElements adjustments
- Advanced case: clear button appearance adjustment for auto-filled inputs

### CODE REVIEW

File: src/app/components/formElements/CheckBoxItem/CheckBoxItem.tsx [hunk 1]
```diff
@@ -1,59 +1,61 @@
...
   label,
   selected,
   type = 'checkbox',
+  disabled,
   onChange,
+  onClick: _onClick,
   ...rest
 }) => {
   const isTypeRadio = type === 'radio';
+  const isypeCheckbox = type === 'checkbox';
```
Comment (mikemike27) [inline src/app/components/formElements/CheckBoxItem/CheckBoxItem.tsx:25]: Hi Salman, just a minor typo here. other than that, everything looks good!

Comment (salman0719) [reply src/app/components/formElements/CheckBoxItem/CheckBoxItem.tsx:25]: @mikemike27 Good catch. Can you please fix it and push? Thanks!

**Review by bhumkarpratik — APPROVED** [review]
LGTM Merge when you want to..
Also it would be better if we can create variables for RGB color values....

**Comment by salman0719** [comment]
> LGTM Merge when you want to.. Also it would be better if we can create variables for RGB color values....

There are several hardcoded color styles all over the application. Need to handle them, creating a new card to address later on.

**Pull request merged**